### DISCUSSION TITLE
Keyboard shortcut for moving the selected tiddler

### DISCUSSION THREAD

**Discussion by arunnbabu81** [BurningTreeC/tiddlywiki-muuri#21] [Ideas]
Can the keyboard shortcut for moving the selected tiddler be added to the current version of muuri like the way it was working in the old [muuri](http://muritest-reloaded.tiddlyspot.com/#Other%20Keyboard%20shortcuts)

Or is it possible to make such a shortcut for moving the selected tiddler like the other shortcuts you have showed in this [demo wiki](http://tiddlywiki-keyboard.tiddlyspot.com/). I am using the keyboard shortcuts shown in this demo in my wikis.

**Comment by BurningTreeC** [comment]
Hi @arunnbabu81 ,
you can create your own keyboard shortcuts for that
to move the currently selected tiddler down:
```
<$set name="currentTiddler" value={{$:/HistoryList!!current-tiddler}}>
<$set name="nextTiddler" value={{{ [list[$:/StoryList]after<currentTiddler>] }}}>
<$action-listops $tiddler="$:/StoryList" $subfilter="+[insertbefore:currentTiddler<nextTiddler>]"/>
</$set>
</$set>
```
to move the currently selected tiddler up:
```
<$set name="currentTiddler" value={{$:/HistoryList!!current-tiddler}}>
<$set name="previousTiddler" value={{{ [list[$:/StoryList]before<currentTiddler>] }}}>
<$action-listops $tiddler="$:/StoryList" $subfilter="+[insertbefore:previousTiddler<currentTiddler>]"/>
</$set>
</$set>
```

**Comment by arunnbabu81** [comment]
Thank you @BurningTreeC . That works. 

Although the moved tiddler is still the active tiddler, it's not in focus within the story river when i am using the krystal horizontal layout. Any idea why it happens like that? All other shortcuts are working as indented as shown in the demo wiki given below.

[Krystal + BTC shortcuts edited demo.zip](https://github.com/BurningTreeC/tiddlywiki-muuri/files/7025271/Krystal.%2B.BTC.shortcuts.edited.demo.zip)

**Reply by BurningTreeC** [reply]
Hi @arunnbabu81 , the keyboard shortcuts I gave you above would probably need to be wrapped in a navigator widget and an `<$action-navigate>` message would need to be sent

**Reply by BurningTreeC** [reply]
for example

```
<$navigator story="$:/StoryList" history="$:/HistoryList">
<$set name="currentTiddler" value={{$:/HistoryList!!current-tiddler}}>
<$set name="nextTiddler" value={{{ [list[$:/StoryList]after<currentTiddler>] }}}>
<$action-listops $tiddler="$:/StoryList" $subfilter="+[insertbefore:currentTiddler<nextTiddler>]"/>
</$set>
</$set>
<$action-navigate $to=<<currentTiddler>>/>
</$navigator>
```

But I haven't tested this and don't know if it works

**Reply by BurningTreeC** [reply]
No, now I've tested this in your wiki and it doesn't work

**Reply by arunnbabu81** [reply]
Thanks for trying to help me. It not an urgent requirement for me.  Do it when ever you get free time only.
